python/ray/dag/input_node.py

```diff
@@ -315,3 +315,35 @@ def __getitem__(self, key: Union[int, str]) -> Any:
                 "Please only use int index or str as first-level key to "
                 "access fields of dag input."
             )
+
+
+class MultiInputNode(DAGNode):
```

Comment:
Can you put this in compiled_dag_node.py instead? It's only used by the compiled DAG code, so best not to put it here.

Reply:
It's also used by channel read to determine if we need to call the get func